src/lib/store.js

```diff
@@ -483,4 +526,49 @@ export default class Store {
 		this.customPurposeList = customPurposeList;
 		this.storeUpdate();
 	};
+
+	mergeVendorConsentsFromGlobalCookie = () => {
+		const selectedLocalIdsOfGlobalVendorInGlobalCookie = new Set();
+		const selectedLocalIdsOfGlobalVendorInLocalCookie = new Set();
+		const globalCookieMaxVendorId = this.persistedGlobalVendorConsentData.maxVendorId;
+
+		this.globalVendorIdsPresentOnList.forEach( (localId, globalId) => {
+			if (this.vendorConsentData.selectedVendorIds.has(localId) && globalId <= globalCookieMaxVendorId) {
+				selectedLocalIdsOfGlobalVendorInLocalCookie.add(localId);
+			}
+			if (this.globalVendorConsentData.selectedVendorIds.has(globalId)) {
+				selectedLocalIdsOfGlobalVendorInGlobalCookie.add(localId);
+				this.vendorConsentData.selectedVendorIds.add(localId);
+			}
+		});
+
+		selectedLocalIdsOfGlobalVendorInLocalCookie.forEach( localId => {
+			const isNewVendorOnList = this.persistedVendorConsentData && localId > this.persistedVendorConsentData.maxVendorId;
+
+			if (!isNewVendorOnList && !selectedLocalIdsOfGlobalVendorInGlobalCookie.has(localId)) {
+				this.vendorConsentData.selectedVendorIds.delete(localId);
+			}
+		});
+	};
+
+	mergeVendorConsentsToGlobalCookie = () => {
+		this.globalVendorIdsPresentOnList.forEach( (localId, globalId) => {
+			if (this.vendorConsentData.selectedVendorIds.has(localId)) {
+				this.globalVendorConsentData.selectedVendorIds.add(globalId);
+			} else {
+				this.globalVendorConsentData.selectedVendorIds.delete(globalId);
+			}
+		});
+	};
+
+	mergePurposeConsentsFromGlobalCookie = () => {
+		const selectedPurposeIds = Array.from(this.vendorConsentData.selectedPurposeIds);
```

Comment:
Array.from is not supported in IE11, please use arrayFrom polyfill which is imported on the top of the file 😁

Reply:
Right 😁